sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample05_IdentityAndSharedAccessCredentials.md

```diff
@@ -0,0 +1,201 @@
+TokenCredential credential = new DefaultAzureCredential();
+
+var storageEndpoint = "<< STORAGE ENDPOINT (likely similar to {your-account}.blob.core.windows.net) >>";
+var blobContainerName = "<< NAME OF THE BLOB CONTAINER >>";
+
+var fullyQualifiedNamespace = "<< NAMESPACE (likely similar to {your-namespace}.servicebus.windows.net) >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+var consumerGroup = "<< NAME OF THE EVENT HUB CONSUMER GROUP >>";
+
+var blobUriBuilder = new BlobUriBuilder(new Uri(storageEndpoint));
+blobUriBuilder.BlobContainerName = blobContainerName;
+
+var storageClient = new BlobContainerClient(
+    blobUriBuilder.ToUri(),
+    credential);
+
+var processor = new EventProcessorClient(
+    storageClient,
+    consumerGroup,
+    fullyQualifiedNamespace,
+    eventHubName,
+    credential);
+
+try
+{
+    using var cancellationSource = new CancellationTokenSource();
+    cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+    // The event handlers are not relevant for this sample; for
+    // illustration, they're delegating the implementation to the
+    // host application.
+
+    processor.ProcessEventAsync += Application.ProcessorEventHandler;
+    processor.ProcessErrorAsync += Application.ProcessorErrorHandler;
+
+    try
+    {
+        await processor.StartProcessingAsync(cancellationSource.Token);
+        await Task.Delay(Timeout.Infinite, cancellationSource.Token);
+    }
+    catch (TaskCanceledException)
+    {
+        // This is expected if the cancellation token is
+        // signaled.
+    }
+    finally
+    {
+        // This may take up to the length of time defined
+        // as part of the configured TryTimeout of the processor;
+        // by default, this is 60 seconds.
+
+        await processor.StopProcessingAsync();
+    }
+}
+catch
+{
+    // If this block is invoked, then something external to the
+    // processor was the source of the exception.
+}
+finally
+{
+   // It is encouraged that you unregister your handlers when you have
+   // finished using the Event Processor to ensure proper cleanup.
+
+   processor.ProcessEventAsync -= Application.ProcessorEventHandler;
+   processor.ProcessErrorAsync -= Application.ProcessorErrorHandler;
+}
+```
+
+## Processing events with Shared Access Signature authorization
+
+**COMING SOON**
+
+## Processing events with Shared Access Key authorization
+
+**COMING SOON**
+
+## Parsing a connection string for information
+
+In some scenarios, it may be preferable to derive token-based authorization from the components of a connection string rather than tracking them separately.  One common scenario for this approach is when your application uses different credentials locally and across the environments where it is hosted.  By using the fully qualified namespace and other values from the connection string, you may be able to reduce duplication and streamline your application's configuration.
```

Comment:
```suggestion
In some scenarios, it may be preferable to supplement token-based authorization with the components of a connection string rather than tracking them separately.  One common scenario for this approach is when your application uses different credentials locally and across the environments where it is hosted.  By using the fully qualified namespace and other values from the connection string, you may be able to reduce duplication and streamline your application's configuration.
```